<launch>
  <!-- yolov3 -->
  <include file="$(find darknet_ros)/launch/yolo_v3.launch">
    <arg name="image" value="/camera/zed_left/image_preprocessed" />
  </include>

  <node name="straight_path_rosbag" pkg="rosbag" type="play" args="--clock /home/ros/data/straight_path_2023_05_17.bag"/>
  <node name="lidar_preprocessing" pkg="perception" type="lidar_preprocessing.py" output="screen"/>
  <node name="camera_preprocessing" pkg="perception" type="camera_preprocessing.py" output="screen"/>

  <!-- static transformations -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="map_to_base_link" args="0 0 0 0 0 0 map base_link" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_hokuyo" args="-0.114 0 0.0674 0 0 0 base_link hokuyo" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_zed_left" args="-0.135 0.053 0.122 0 0 0 base_link zed_left" />
</launch>